lib/sles4sap_publiccloud.pm

```diff
@@ -608,4 +610,48 @@ sub delete_network_peering {
     }
 }
 
+=head2 get_hana_database_info
+
+    Parses "hdbsql" command output.
+    Returns true if hana database is online.
+
+=cut
+
+sub get_hana_database_info {
+    my ($self) = @_;
+    my $result = -1;
+    my $output_cmd = $self->run_cmd(cmd => "hdbsql -u SYSTEM -p Linux1234 -i 00 'SELECT * FROM SYS.M_DATABASES;'", runas => get_required_var("SAP_SIDADM"));
+    my $hana_database_online = '"HDB","HDB-00","YES","ACTIVE","","","DEFAULT","?"';
+
+    $result = ($output_cmd =~ /$hana_database_online/ ? 1 : 0);
+    return $result;
+}
+
+=head2 is_hana_database_offline
+
+    $self->is_hana_database_offline();
```

Comment:
why these two examples with and without the `self` ?

Reply:
Please could you explain me better your point?